docs/docs-beta/docs/dagster-plus/deployment/hybrid/agents/kubernetes.md

```diff
@@ -113,14 +138,315 @@ helm --namespace dagster-cloud upgrade agent \
+
+
+</TabItem>
+</Tabs>
+
+:::note
+If you need to request secrets from a secret manager like AWS Secrets Manager or Hashicorp Vault, follow one of the prior methods to give your code access to vault credentials. Then, inside your Dagster code, use those  credentials to authenticate a Python client that requests secrets from the secret manager.
+:::
+
+
+### Use a different service account for a specific code location
+
+Modify the [`dagster_cloud.yaml` file](/todo) in your project's Git repository:
+
+```yaml file=dagster_cloud.yaml
+locations:
+  - location_name: cloud-examples
+    image: dagster/dagster-cloud-examples:latest
+    code_source:
+      package_name: dagster_cloud_examples
+    container_context:
+      k8s:
+        service_account_name: my_service_account_name
+```
+
+### Run Dagster+ with different Kubernetes clusters
+
+<Tabs>
+<TabItem value="ha" label="Work can happen in either cluster">
+
+Deploy the agent Helm chart to each cluster, setting the `isolatedAgents.enabled` flag to true.
+
+```yaml file=values.yaml
+# values.yaml
+isolatedAgents:
+  enabled: true
+```
+
+```shell
+helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --values ./values.yaml
+```
+
+In this configuration, requests will be randomly distributed across agents in both clusters.
+
+</TabItem>
+<TabItem value="queues" label="Global asset graph, but projects run in different clusters">
+
+You may wish to run data pipelines from project A in Kubernetes cluster A, and project B in Kubernetes cluster B. For example, you may wish to run some jobs on-premise and other jobs in the cloud. To accomplish this task:
+
+- Deploy an agent into each environment and use the `agentQueue` configuration:
+  Specify an agent queue for the on-prem agent:
+  ```yaml file=values.yaml
+  dagsterCloud:
+    agentQueues:
+      additionalQueues:
+        - on-prem-agent-queue
+  ```
+  Deploy onto the on-prem Kubernetes cluster:
+  ```shell
+  helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --values ./values.yaml
+  ```
+  Specify an agent queue for the cloud agent:
+
+  ```yaml file=values.yaml
+  dagsterCloud:
+    agentQueues:
+      additionalQueues:
+        - cloud-agent-queue
+  ```
+  Deploy onto the cloud Kubernetes cluster:
+  ```shell
+  helm --namespace dagster-cloud upgrade agent \
+    dagster-cloud/dagster-cloud-agent \
+    --values ./values.yaml
+  ```
+- Create separate code locations for each project
+- Update the `dagster_cloud.yaml` file for each code location
+  ```yaml file=dagster_cloud.yaml
+  locations:
+    - location_name: project-a
+      ...
+      agent_queue: on-prem-agent-queue
+    - location_name: project-b
+      ...
+      agent_queue: cloud-agent-queue
+  ```
+
+:::tip
+Code locations without an `agent_queue` will be routed to a default queue. You pick an agent to serve this default queue using the `includeDefaultQueue` setting:
```

Comment:
note that this is true by default - you have to explicitly turn it off if you don't want your agent to serve the default queue, even if you specify additionalQueues